### PR TITLE
feat: added a loader for StarSearch responses

### LINKED ISSUE #3203: Feature: better loading state
### Current behavior

We don't have a loading state right now, and when the info comes in, it just replicates the stream flow from the API.

### Suggested solution

It'd be great to have a nicer animation for the loading state and prompt reply creation.

### Inspiration
Rewatch GIF Recording - 2024-04-17 at 11 19 18 AM

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by nickytonline**
## Description

Now StarSearch has a loading animation before a response is streamed in.

## Related Tickets & Documents

Close #3203

## Mobile & Desktop Screenshots/Recordings

**Before**

CleanShot 2024-05-17 at 18 40 14

**After**

CleanShot 2024-05-17 at 23 58 52

## Steps to QA

1. Go to StarSearch
2. Click on one of the canned prompts
3. Notice the three dots loader while an anwer is streamed in.

## Tier (staff will fill in)

- [ ] Tier 1
- [ ] Tier 2
- [x] Tier 3
- [ ] Tier 4

### CODE REVIEW

**Comment by bdougie** [comment]
Feels off. I'd rather figure out how to remove loader when component renders or look for a different solution. We can chat through this more next week.

**Comment by nickytonline** [comment]
> Feels off. I'd rather figure out how to remove loader when component renders or look for a different solution. We can chat through this more next week.

This was bugging me so I fixed it. Now the loader appears when a new prompt is sent and it disappears as soon as a valid response message returns.

File: tailwind.config.js [hunk 1]
```diff
@@ -36,7 +36,7 @@ module.exports = {
       },
       colors: {
         sauced: {
-          orange: "hsla(19, 100%, 50%, 1)",
+          orange: "hsla(19, 100%, 50%, var(--tw-bg-opacity))",
```
Comment (nickytonline) [inline tailwind.config.js:39]: This was to allow me to set the opacity via `bg-opacity-50` but it unlocks this colour now to be able to set an opacity on it. Prior to this, it was always 100% opacity.

File: pages/star-search/index.tsx [hunk 2]
```diff
@@ -287,7 +288,7 @@ export default function StarSearchPage({ userId, bearerToken, ogImageUrl }: Star
       const { done, value } = await reader!.read();
       if (done) {
         setIsRunning(false); // enables input
-        const textChat = chat.filter((item) => typeof item.content === "string");
```
Comment (nickytonline) [inline pages/star-search/index.tsx:290]: This was an unused variable from a previous PR.

File: pages/star-search/index.tsx [hunk 3]
```diff
@@ -432,16 +433,36 @@ export default function StarSearchPage({ userId, bearerToken, ogImageUrl }: Star
           </div>
         );
       case "chat":
+        // We only want to process the chat messages that are either strings or valid widgets.
+        // The API currently sends back other function calls that we currently do not support or don't need to support,
+        // so we filter those out by checking if they are in the component registry.
+        const chatMessagesToProcess = chat.filter(
+          (c) => typeof c.content === "string" || componentRegistry.has(c.content.name)
+        );
```
Comment (nickytonline) [inline pages/star-search/index.tsx:441]: Once #3424 is merged, this filtering can be removed and we can filter out immediately when a message comes in that is a `function_call` where the `name` in the JSON payload isn't in the component registry.

**Review by bdougie — APPROVED** [review]
Works.

**Review by BekahHW — APPROVED** [review]

**Review by BekahHW — APPROVED** [review]

**Pull request merged**